application/modules/statements/controllers/Statements.php

```diff
@@ -0,0 +1,384 @@
+
+            $transaction = [
+                'transaction_type'          => self::TRANSACTION_TYPE_PAYMENT,
+                'transaction_date'          => $payment_entry->payment_date,
+                'transaction_amount'        => $payment_entry->payment_amount,
+
+
+                'invoice_id'                => $payment_entry->invoice_id,
+                'client_id'                 => $payment_entry->client_id,
+                'invoice_date_created'      => $payment_entry->invoice_date_created,
+                'invoice_item_subtotal'     => $payment_entry->invoice_item_subtotal,
+                'invoice_item_tax_total'    => $payment_entry->invoice_item_tax_total,
+                'invoice_total'             => $payment_entry->invoice_total,
+                'invoice_sign'              => $payment_entry->invoice_sign,
+                'invoice_number'            => $payment_entry->invoice_number,
+                'payment_id'                => $payment_entry->payment_id,
+                'payment_method_id'         => $payment_entry->payment_method_id,
+                'payment_method_name'       => $payment_entry->payment_method_name,
+                'payment_date'              => $payment_entry->payment_date,
+                'payment_amount'            => $payment_entry->payment_amount,
+
+            ];
+
+            $statement_transactions[] = $transaction;
+
+            $client_total_balance -= $payment_entry->payment_amount;
+
+        }
+
+        usort($statement_transactions, array($this, "compare_statement_dates"));
+
+        $this->mdl_statement->setStatement_transactions($statement_transactions);
+
+        $this->mdl_statement->setStatement_balance($client_total_balance);
+
+
+        return $this->mdl_statement;
+
+    }
+
+
+
+
+    /**
+     */
+    public function generate_pdf()
+    {
+
+        $client_id              = $this->input->post('cid');
+        $statement_number       = $this->input->post('statement_number');
+
+        if (!empty($this->input->post('statement_start_date'))) {
+            $statement_start_date   = strtotime($this->input->post('statement_start_date'));
+        } else {
+            $statement_start_date   = strtotime($this->input->post('sdate'));
+        }
+        $statement_end_date     = strtotime($this->input->post('edate'));
+
+        $statement_date         = $this->input->post('statement_date_created');
+        $notes                  = $this->input->post('notes');
+
+
+        $this->generate_statement_pdf($client_id, $statement_number, $statement_start_date, $statement_end_date, $statement_date, $notes);
+    }
+
+    /**
+     * Generate the PDF for the statement
+     *
+     * @param $quote_id
+     * @param bool $stream
+     * @param null $quote_template
+     *
+     * @return string
+     * @throws \Mpdf\MpdfException
+     */
+    function generate_statement_pdf($client_id, $statement_number, $statement_start_date, $statement_end_date, $statement_date, $notes)
```

Comment:
@pradeshc Please move this function into the [helpers/mpdf_helper.php](https://github.com/InvoicePlane/InvoicePlane/blob/master/application/helpers/mpdf_helper.php) file